jaeger-thrift/src/main/java/io/jaegertracing/thrift/internal/senders/HttpSender.java

```diff
@@ -119,13 +134,98 @@ public Builder withAuth(String authToken) {
       return this;
     }
 
+    public Builder withCertificatePinning(String[] sha256certs /* comma separated */) {
+      pins.addAll(Arrays.asList(sha256certs));
+      return this;
+    }
+
+    /**
+     * Enable accepting self-signed certificates. This will only take effect if pinning is provided.
+     */
+    public Builder acceptSelfSigned() {
+      this.selfSigned = true;
+      return this;
+    }
+
     public HttpSender build() {
       if (authInterceptor != null) {
         clientBuilder.addInterceptor(authInterceptor);
       }
+      try {
+        // Just obtain hostname for SSL feature. Failure is OK if plain http is used.
+        final URI uri = new URI(endpoint);
+        String hostname = hostname = uri.getHost();
+
+        if (!selfSigned && !pins.isEmpty()) {
+          // Pinning Certificate issued by public CA
+          for (String cert: pins) {
+            certificatePinnerBuilder.add(hostname, String.format("%s", cert));
+          }
+          clientBuilder.certificatePinner(certificatePinnerBuilder.build());
+        } else if (selfSigned && !pins.isEmpty()) {
+          /* Issued by private CA, OkHttp's pinner is unable to verify the pins.
+           * Instead, check the sha256 hash value by custom verifier. */
+          acceptSelfSigned(clientBuilder, hostname, pins);
+        }
+      } catch (Exception e) {
```

Comment:
This deserves a log message, even if at `debug` level (but possibly at info or even warn, as this is called only once)

Reply:
I afraid it isn't, because it is very similar to https://github.com/jaegertracing/jaeger-client-java/pull/602/files/a0607f48cd7370959b8f937a208fdc517a03b82f#diff-3330f8a6420c943a90a1b22b0abd0815R56. But it might be OK to fail earlier than build itself.

Reply:
Not sure I understood what you mean. The code you linked throw an exception, the code here is swallowing one. Instead of swallowing, we should log it at some level (`info` in my opinion).

Reply:
The `try` above the `catch` is needed because of `URI` to be constructed by the `endpoint` which is supplied by the configuration. If the endpoint is invalid, building `HttpSender` at few lines later would fail. But I agree with that it shouldn't swallow all kind of `Exception`, so I changed it to catch only `URISyntaxException` and throw it earlier.